suplemon/viewer.py

```diff
@@ -621,3 +604,251 @@ def purge_line_cursors(self, line_no):
         for line_cursors in cursor:
             self.remove_cursor(cursor)
         return True
+
+    ###########################################################################
+    # Input Handling
+    ###########################################################################
+
+    def get_key_bindings(self):
+        """Get list of editor key bindings."""
+        return self.app.get_key_bindings()
+
+    def handle_input(self, event):
+        """Handle input."""
+        if event.type == "mouse":
+            return False
+        key = event.key_code
+        name = event.key_name
+        # Try match a key to a method and call it
+
+        key_bindings = self.get_key_bindings()
+        operation = None
+        if key in key_bindings.keys():
+            operation = key_bindings[key]
+        elif name in key_bindings.keys():
+            operation = key_bindings[name]
+        if operation:
+            self.run_operation(operation)
+            return True
+        return False
+
+    def run_operation(self, operation):
+        """Run an editor core operation."""
+        if operation in self.operations.keys():
+            cancel = self.app.trigger_event_before(operation)
+            if cancel:
+                return False
+            result = self.operations[operation]()
+            self.app.trigger_event_after(operation)
+            return result
+        return False
+
+    ###########################################################################
+    # Operations
+    ###########################################################################
+
+    def arrow_right(self):
+        """Move cursors right."""
+        for cursor in self.cursors:
+            line = self.lines[cursor.y]
+            if cursor.y != len(self.lines)-1 and (cursor.x >= len(line) or len(line) == 0):
+                cursor.move_down()
+                cursor.set_x(0)
+            elif cursor.x < len(self.lines[cursor.y]) and len(line) > 0:
+                cursor.move_right()
+        self.move_cursors()
+        self.scroll_down()
+
+    def arrow_left(self):
+        """Move cursors left."""
+        for cursor in self.cursors:
+            if cursor.y != 0 and cursor.x == 0:
+                cursor.move_up()
+                cursor.set_x(len(self.lines[cursor.y])+1)
+        self.move_cursors((-1, 0))
+        self.scroll_up()
```

Comment:
I must admit that the sublime way looks very nice and that's how I'd like it to work in suplemon too. This requres some more refactoring in many places so I'll postpone this for later. I'll try to finalize and :shipit: the bug fixes and new stuff in this PR first.

Thanks for the suggestion, I'll definitely come back to this when I have time.